src/gui/updateurldialog.cpp

```diff
@@ -0,0 +1,60 @@
+#include "updateurldialog.h"
+
+#include <QMessageBox>
+// this include allows comparing QAbstractButton* and QPushButton*
+#include <QPushButton>
+
+namespace OCC {
+
+UpdateUrlDialog::UpdateUrlDialog(const QString &messageBoxTitle, const QString &messageBoxContent, const QUrl &oldUrl, const QUrl &newUrl, QWidget *parent)
```

Comment:
We do we inherit QObject and not QDialog?

Reply:
Using `QWidget` allows passing the parent to the message box, which expects such a pointer.

Reply:
I mean why do you not inherit QDialog or QMessagebox but build that wrapper around it?

Reply:
See https://github.com/owncloud/client/pull/9422/files/fb490599e3ca47b86799f6de8ddd3231a3ba1b8a#r800696137.